R/layout.R

```diff
@@ -258,33 +258,30 @@ Layout <- ggproto("Layout", NULL,
     default <- unclass(plot$mapping)
 
     self$layer_mappings <- lapply(plot$layers, function(layer) {
-      mapping <- default
-      default[names(layer$mapping)] <- layer$mapping
-      mapping
+      defaults(layer$mapping, default)
```

Comment:
I don't think so, because the `defaults` function gets passed `layer$mapping`, not `layer`.